__tests__/helpers/pairs.spec.ts

```diff
@@ -1,13 +1,99 @@
-import { parsePairs } from "../../src/helpers/pairs";
+import {
+  parsePairs,
+  getInterpolatorForPair,
+  Interpolator,
+} from '../../src/helpers/pairs';
+import { rgba as RGBA } from '../../src/interpolate';
 
-describe("pairs", () => {
-  describe("parsePairs", () => {
-    it("should parse a from / to pair to its property and value representations", () => {
+describe('pairs', () => {
+  describe('parsePairs', () => {
+    it('should parse a from / to pair to its property and value representations', () => {
       const pair = { from: { opacity: 0 }, to: { opacity: 1 } };
       expect(parsePairs(pair)).toEqual({
-        from: { property: "opacity", value: 0 },
-        to: { property: "opacity", value: 1 }
+        from: { property: 'opacity', value: 0 },
+        to: { property: 'opacity', value: 1 },
       });
     });
   });
+
+  describe('getInterpolatorForPair', () => {
+    it('should infer a from / to pair of type number and return a numeric interpolator', () => {
+      const pair = { from: { opacity: 0 }, to: { opacity: 1 } };
+      const { interpolator } = getInterpolatorForPair(pair);
+
+      const result = interpolator({
+        range: [100, 400],
+        domain: [0.3, 0.5],
+        value: 175,
+      });
+      expect(result).toEqual(0.35);
+    });
+
+    it('should infer a from / to pair of type string that matches a color and return a color interpolator', () => {
+      const pair = {
+        from: { background: '#c86432bf' }, // rgba(200, 100, 50, 0.75)
+        to: { background: '#64c84b' }, // rgba(100, 200, 75, 1)
+      };
+      const { interpolator } = (getInterpolatorForPair(pair) as unknown) as {
+        interpolator: Interpolator<RGBA<number>, string>;
+      };
```

Comment:
In truly lame fashion, TS doesn't seem to be properly inferring the overloads on this function based on how it's used – good ol' ISTS. If anyone is an expert in getting TS overloads to play nicely, definitely hit me up!